starlark/eval.go

```diff
@@ -1659,8 +1659,10 @@ func interpolate(format string, x Value) (Value, error) {
 		index++
 	}
 
-	if index < nargs {
-		return nil, fmt.Errorf("too many arguments for format string")
+	if _, ok := x.(Mapping); !ok {
```

Comment:
Let's check for Mapping only in the uncommon case:
```
if index < nargs && !is[Mapping](x) { ... }

func is[T any](x T) bool {
    _, ok := x.(T)
    return ok
}
```

Reply:
Sounds good, I'm just adding this. There's an unfortunate name clash with the `any` builtin, would it be better to rename this to `any_` or to use `interface{}` for `is` as I see more of the pre-1.18 in the codebase?

Reply:
> There's an unfortunate name clash with the any builtin, would it be better to rename this to any_ or to use interface{} for is as I see more of the pre-1.18 in the codebase?

What do you mean by name clash? The `is` function I suggested above (which should be placed at top level at the end of the file) intentionally refers to the `any` built-in. (You could say interface{} but that is almost never necessary since `any` is in scope nearly everywhere.)

Reply:
Ah I think we're referring to different things. I was referring to the implementation of Starlark's built-in `any(iter)` function. Currently, I get the following compilation error on `is`:
```
starlark/eval.go:1669:11: any (value of type func(thread *Thread, _ *Builtin, args Tuple, kwargs []Tuple) (Value, error)) not a type
```
My question is whether the implementation function for Starlark's `any()` should be renamed to allow Go's `any` built-in to appear in the signature of `is` however I don't see any use of Go's `any` in the codebase at the moment